WalletWasabi.Gui/Global.cs

```diff
@@ -247,6 +247,16 @@ public async Task InitializeNoWalletAsync(TerminateService terminateService)
 					BitcoinStore.BlockRepository);
 
 				WalletManager.RegisterServices(BitcoinStore, Synchronizer, Config.ServiceConfiguration, HostedServices.Get<HybridFeeProvider>(), blockProvider);
+
+				try
+				{
+					StoppingCts.CancelAfter(10000);
```

Comment:
- Create a new CancellationTokesSource() inside try/c with timeout of 10 seconds. 
- Create a LinkedCancellationTokenSource and link them together with StoppingCts. 
- Pass the linked token

So it will be either canceled by the StoppingCts or the timeout

Reply:
I see now. Thanks!